msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/TestHelper.java

```diff
@@ -27,4 +46,55 @@ static void deleteFileContent(Class<?> classInstance, String resource)
         fileWriter.write("");
         fileWriter.close();
     }
+
+    static String generateToken() {
+        try {
+            RSAKey rsaJWK = new RSAKeyGenerator(2048)
+                    .keyID("kid")
+                    .generate();
+            JWSObject jwsObject = new JWSObject(
+                    new JWSHeader.Builder(JWSAlgorithm.RS256).keyID(rsaJWK.getKeyID()).build(),
+                    new Payload("payload"));
+
+            jwsObject.sign(new RSASSASigner(rsaJWK));
+
+            return jwsObject.serialize();
+        } catch (JOSEException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    //Maps various values to the successfulResponseFormat string to create a valid token response
+    static String getSuccessfulTokenResponse(HashMap<String, String> responseValues) {
+        //Will default to expiring in one hour if expiry time values are not set
+        long expiresIn = responseValues.containsKey("expires_in") ?
+                Long.parseLong(responseValues.get("expires_in")) :
+                3600;
+        long expiresOn = responseValues.containsKey("expires_on")
+                ? Long.parseLong(responseValues.get("expires_0n")) :
+                (System.currentTimeMillis() / 1000) + expiresIn;
+
+        return String.format(successfulResponseFormat,
+                responseValues.getOrDefault("access_token", "access_token"),
+                responseValues.getOrDefault("id_token", "id_token"),
```

Comment:
Hmm, does this just work? MSAL does parse the id token to get details such as `tid` etc.

Reply:
Yes it just works, though it probably shouldn't. As we talked about yesterday we rely on that nimbus library, and you can see here we pass the values into their "OIDCTokens" method: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/95b5efc9fc30514719e5b678c739351ddcf35fba/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenResponse.java#L30

That doesn't through any error when we pass in something like "id_token". Instead, the instance of TokenResponse that gets made has a getOIDCTokens() method which has two relevant fields:
 - `idTokenString`, which is the actual string we give it ("id_token" in this case) 
 - `idToken`, which is JWT object that is simply null if we pass in something like "id_token"

And then later on in the library it seems like having a 'null' for `idToken` means that everything acts like it's a confidential client flow that had no ID token and nothing breaks. Based on what we talked about before I just created an issue about using those packages less: https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/889